dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -516,6 +609,66 @@ def _perform_t_test(
         results["welch"]["p-value"] = float(welch_p_val)
         return results
 
+    @staticmethod
+    def _calculate_psi(
+        self_match_count: int,
+        histogram1: np.ndarray,
+        other_match_count: int,
+        histogram2: np.ndarray,
+    ) -> Optional[float]:
+        """
+        Calculate PSI (Population Stability Index).
+
+        ```
+        PSI = SUM((other_pcnt - self_pcnt) * ln(other_pcnt / self_pcnt))
+        ```
+
+        PSI Breakpoint Thresholds:
+            - PSI < 0.1: no significant population change
+            - 0.1 < PSI < 0.2: moderate population change
+            - PSI >= 0.2: significant population change
+
+        :param self_match_count: self.match_count
+        :type self_match_count: int
+        :param histogram1: self._stored_histogram["histogram"]
+        :type histogram1: np.ndarray
+        :param self_match_count: other_profile.match_count
+        :type self_match_count: int
+        :param histogram2: other_profile._stored_histogram["histogram"]
+        :type histogram2: np.ndarray
+        :return: psi_value
+        :rtype: optional[float]
+        """
+        psi_value = 0
+        invalid_stats = False
+
+        if isinstance(histogram2["bin_edges"], type(None)) or isinstance(
+            histogram1["bin_edges"], type(None)
+        ):
+            warnings.warn(
+                "No edges available in at least one histogram for calculating `PSI`",
+                RuntimeWarning,
+            )
+            invalid_stats = True
+
+        if invalid_stats:
```

Comment:
Instead of an extra if could we put it inside the previous?

Reply:
lol true 🤦

Reply:
fixed in next push